### ISSUE TITLE
[Feature Request] Globally-defined logging to artifact repository

### ISSUE THREAD

**Issue by irlevesque** [argoproj/argo#1790]
## Feature Request
As an administrator, I'd like to ensure that all cluster users' workflow logs are always persisted to our globally-defined artifact repository.

## Problem
Currently, Argo requires Workflow developers to remember to add `archiveLocation.archiveLogs: true` to every template. This is error prone and doesn't provide compliance guarantees.

## Solution
Provide a global logging option in the controller ConfigMap that enables `archiveLogs` globally, regardless of Workflow `archiveLocation` specification.

## Refs
[Slack discussion](https://argoproj.slack.com/archives/C8J6SGN12/p1574443372000200)

**Comment by alexec** [comment]
This already partially works

Add this to your config map:

```
    # artifactRepository defines the default location to be used as the artifact repository for
    # container artifacts.
    artifactRepository:
      # archiveLogs will archive the main container logs as an artifact
      archiveLogs: true
```

Anything that uses uses artifacts will also archive logs.

If they do not use it, they do not archive logs.

**Comment by simster7** [comment]
Also fixed with #2331
